### PR TITLE
Fix bug in SmtLibWriter

### PR DESCRIPTION
**Pull request by OmriIsacHUJI**

### CODE REVIEW

File: src/common/MString.cpp [hunk 1]
```diff
@@ -175,6 +175,23 @@ String String::trim() const
     return substring( firstNonSpace, lastNonSpace - firstNonSpace + 1 );
 }
 
+String String::trimZerosFromRight() const
```
Comment (guykatzz) [inline src/common/MString.cpp:178]: Wouldn't it make more sense to traverse the string from the right?

File: src/proofs/SmtLibWriter.cpp [hunk 2]
```diff
@@ -91,5 +92,8 @@ void SmtLibWriter::writeInstanceToFile( IFile &file, const List<String> &instanc
 
 String SmtLibWriter::signedValue( double val )
 {
-    return val > 0 ? std::to_string( val ) : "( - " + std::to_string( abs( val ) ) + " )";
+    unsigned precision = ( unsigned ) std::log10 ( 1 / GlobalConfiguration::DEFAULT_EPSILON_FOR_COMPARISONS );
+    std::stringstream s;
+    s << std::fixed << std::setprecision( precision ) << abs( val );
+    return val >= 0 ? String ( s.str() ).trimZerosFromRight() : String( "( - " + s.str() ).trimZerosFromRight() + " )";
```
Comment (guykatzz) [inline src/proofs/SmtLibWriter.cpp:98]: No space between a function name and the ensuing (
String ( s.str() --> String( s.str()
Likewise with std::log10 ( 1 ...

**Review by guykatzz — COMMENTED** [review]
Also, I believe this PR fails the integration tests

File: src/common/MString.cpp (hunk 1 above)
Comment (omriisack) [reply src/common/MString.cpp:178]: It will. Fixed.

**Review by guykatzz — APPROVED** [review]

**Pull request merged**